pyigm/cgm/cgm.py

```diff
@@ -335,35 +358,53 @@ def stack_plot(self,to_plot,pvlim=None,maxtrans=3,return_fig=True,**kwargs):
                         # Load spectrum if not already loaded
                         if al.analy['spec'] is None:
                             try:
-                              al.analy['spec'] = readspec(al.analy['spec_file'])
+                                if spec is not None:
+                                    al.analy['spec'] = spec
+                                else:
+                                    al.analy['spec'] = readspec(al.analy['spec_file'])
                             except:
-                              raise LookupError("analy['specfile'] must be "
-                                                "declared for AbsLines")
+                                raise LookupError("spec must be defined or "
+                                                  "analy['specfile'] must be "
+                                                  "declared for AbsLines")
                         # Get wavelength limits to know where to look
                         wmins.append(al.analy['spec'].wvmin.value)
                         wmaxs.append(al.analy['spec'].wvmax.value)
                     wlims= (np.min(np.array(wmins)),np.max(np.array(wmaxs)))*u.Angstrom
                     # ID the strong transitions
                     strong = ilist.strongest_transitions(
-                                tp,wvlims=wlims/(1.+comp.zcomp),n_max=maxtrans)
+                        tp,wvlims=wlims/(1.+comp.zcomp),n_max=maxtrans)
                     if strong is None: #  No lines covered in the spectra
                         warnings.warn('No lines for {} are covered by the spectra'
                                       'provided.'.format(tp))
                         continue
                     # Grab the AbsLines from this AbsComponent and their names
                     complines = comp._abslines
+                    complines = np.array(complines) # For the indexing
                     compnames = np.array([ll.name for ll in complines])
                     ### Add the lines found to the master list
                     if isinstance(strong,dict):  # Only one line covered
-                        lines2plot.append(complines[compnames == strong['name']])
+                        lines2plot.append(complines[compnames == strong['name']][0])
                     else:  # Multiple lines covered
-                        complines = np.array(complines) # For the indexing
-                        try:
-                            tokeep = [complines[compnames == sn][0] for sn in strong['name']]
-                            lines2plot.extend(tokeep)
-                        except:
-                            warnings.warn('{} covered by spectra but not in'
-                                          'components list'.format(sn))
+                        for i,sn in enumerate(strong['name']):
+                            if sn in compnames:
+                                tokeep = [complines[compnames == sn][0]]
```

Comment:
Add a few comments in here if you can.
Thanks!

Reply:
Done.